### PR TITLE
deps: Experiment — adopt canonical CEL - py_proto_library support via rules_python

### PR DESCRIPTION
**Pull request by sergiitk**
DO NOT MERGE

This is an experiment at this moment.

Commit Message:
Additional Description:
Risk Level:
Testing:
Docs Changes:
Release Notes:
Platform Specific Features:
[Optional Runtime guard:]
[Optional Fixes #Issue]
[Optional Fixes commit #PR or SHA]
[Optional Deprecated:]
[Optional [API Considerations](https://github.com/envoyproxy/envoy/blob/main/api/review_checklist.md):]

### CODE REVIEW

File: api/bazel/repository_locations.bzl [hunk 1]
```diff
@@ -39,11 +39,11 @@ REPOSITORY_LOCATIONS_SPEC = dict(
         project_desc = "xDS API Working Group (xDS-WG)",
         project_url = "https://github.com/cncf/xds",
         # During the UDPA -> xDS migration, we aren't working with releases.
-        version = "e9ce68804cb4e64cab5a52e3c8baf840d4ff87b7",
-        sha256 = "0d33b83f8c6368954e72e7785539f0d272a8aba2f6e2e336ed15fd1514bc9899",
+        version = "expression-new-fields",
```
Comment (htuch) [inline api/bazel/repository_locations.bzl:42]: Let me know when ready for review. It looks like this is still pointing at a branch.

Comment (sergiitk) [reply api/bazel/repository_locations.bzl:42]: Yes - this is still a draft. I'll click "Ready for review" when it's ready.

**Comment by moderation** [comment]
@phlax, @htuch at some point it would be nice to repoint to the correct repo - https://github.com/cncf/xds - and change all the references to `udpa`. There are a _lot_ of references to the incorrect repo - https://github.com/search?q=repo%3Aenvoyproxy%2Fenvoy+com_github_cncf_udpa&type=code

```tools/api_proto_plugin/plugin.bzl
40:            f.path.startswith("external/com_github_cncf_udpa/xds"))

tools/protoxform/protoxform.bzl
27:                path.short_path.startswith("../com_github_cncf_udpa") or

tools/protodoc/protodoc.bzl
23:                path.short_path.startswith("../com_github_cncf_udpa")

api/bazel/repositories.bzl
29:        name = "com_github_cncf_udpa",

api/bazel/external_proto_deps.bzl
25:    # @com_github_cncf_udpa//xds/type/matcher/v3:pkg_go_proto

api/bazel/repository_locations.bzl
37:    com_github_cncf_udpa = dict(
```

**Comment by sergiitk** [comment]
Superseded by another approach: https://github.com/envoyproxy/envoy/pull/30834.

**Pull request closed without merging**